lib/ansible/modules/system/systemd.py

```diff
@@ -457,6 +457,12 @@ def main():
                         (rc, out, err) = module.run_command("%s %s '%s'" % (systemctl, action, unit))
                         if rc != 0:
                             module.fail_json(msg="Unable to %s service %s: %s" % (action, unit, err))
+
+                    # Test real state after systemctl start
+                    (rc, out, err) = module.run_command("%s status '%s'" % (systemctl, unit))
+                    if action == "start":
```

Comment:
I would imagine the same issue is possible for other states also, i would solve this in a generic way instead of only for one of them.